src/main/java/net/shadowmage/ancientwarfare/structure/render/ProtectionFlagRenderer.java

```diff
@@ -108,39 +105,23 @@ private void render(float x, float y, float z, float partialTicks, float alpha,
 		float f3 = (float) (pos.getX() * 7 + pos.getY() * 9 + pos.getZ() * 13) + worldTime + partialTicks;
 		bannerModel.bannerSlate.rotateAngleX = (-0.0125F + 0.01F * MathHelper.cos(f3 * (float) Math.PI * 0.02F)) * (float) Math.PI;
 		GlStateManager.enableRescaleNormal();
-		ResourceLocation resourcelocation = getBannerResourceLocation(topColor, bottomColor);
-
-		if (resourcelocation != null) {
-			Minecraft.getMinecraft().renderEngine.bindTexture(resourcelocation);
-			GlStateManager.pushMatrix();
-			GlStateManager.scale(0.6666667F, -0.6666667F, -0.6666667F);
-			bannerModel.renderBanner();
-			GlStateManager.popMatrix();
-		}
+		ResourceLocation resourcelocation = getBannerResourceLocation(topColor, bottomColor, faction);
+
+		Minecraft.getMinecraft().renderEngine.bindTexture(resourcelocation);
+		GlStateManager.pushMatrix();
+		GlStateManager.scale(0.6666667F, -0.6666667F, -0.6666667F);
+		bannerModel.renderBanner();
+		GlStateManager.popMatrix();
 
 		GlStateManager.color(1.0F, 1.0F, 1.0F, alpha);
 		GlStateManager.popMatrix();
 	}
 
-	private ResourceLocation getBannerResourceLocation(int topColor, int bottomColor) {
-		String id = getBannerId(topColor, bottomColor);
-		ResourceLocation textureEntry = flagTextures.get(id);
-		if (textureEntry == null) {
-			List<String> list = new ArrayList<>();
-			list.add("textures/entity/banner/base.png");
-			list.add("textures/entity/banner/half_horizontal_bottom.png");
-			textureEntry = new ResourceLocation(id);
-			Minecraft.getMinecraft().getTextureManager().loadTexture(textureEntry, new LayeredCustomColorMaskTexture(BASE_BANNER_TEXTURE, list, ImmutableList.of(topColor, bottomColor)));
-			flagTextures.put(id, new ResourceLocation(id));
-		}
-
+	private ResourceLocation getBannerResourceLocation(int topColor, int bottomColor, String faction) {
+		ResourceLocation textureEntry = new ResourceLocation(AncientWarfareCore.MOD_ID, "textures/entity/structure/banner/" + faction + ".png");
```

Comment:
This should be a constant if it's supposed to be constant so that we don't need to create a new instance every single time this renders which is every single tick.